scylla/src/policies/speculative_execution.rs

```diff
@@ -180,6 +180,11 @@ where
                     } else {
                         last_error = Some(r)
                     }
+                } else {
+                    // The only case where None is returned is when execution plan was exhausted.
+                    // If so, there is no reason to start any more fibers.
+                    // We can't always return - there may still be fibers running.
+                    retries_remaining = 0;
                 }
                 if async_tasks.is_empty() && retries_remaining == 0 {
                     return last_error.unwrap_or({
```

Comment:
:question: Does it make sense to just finish this block of code if `async_tasks.is_empty()` but `retries_remaining > 0`? In such case we're going to unproductively wait for a new fiber to be spawn once the timer fires, or am I mistaken?
:thought_balloon: Actually, `async_tasks.is_empty()` means that there are no other running fibers. So if there are any more fibers that could be started, I guess we should spawn one immediately. WDYT?

Reply:
Waiting may make sense - for example if cluster is overloaded. We even have an issue open about adding waiting functionality to RetryPolicy, and we had in-person discussions about that in summit. Imo removing waiting here may be problematic in that context, and also surprising to users (driver would not respect retry_interval).
In any case, it is not related to this PR. If my arguments are not convincing, please open an issue and let's discuss it further - maybe your idea is good and I just don't see it yet.